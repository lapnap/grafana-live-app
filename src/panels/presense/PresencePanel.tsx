// Libraries
import React, {PureComponent} from 'react';

// Types
import {PresenseOptions} from './types';
import {PanelProps} from '@grafana/ui';

export interface Props extends PanelProps<PresenseOptions> {}

interface State {
  html: string;
}

export class PresencePanel extends PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      html: 'xxxx',
    };
  }

  render() {
    return <div>TODO! presense panel!!!</div>;
  }
}
