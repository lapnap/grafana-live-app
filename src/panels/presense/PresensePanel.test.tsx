import React from 'react';
import {shallow} from 'enzyme';
import {PresencePanel, Props} from './PresencePanel';

const setup = (propOverrides?: object) => {
  const props: Props = {
    height: 300,
    width: 300,
  } as Props; // partial

  Object.assign(props, propOverrides);

  const wrapper = shallow(<PresencePanel {...props} />);
  const instance = wrapper.instance() as PresencePanel;

  return {
    instance,
    wrapper,
  };
};

describe('Render Presense Panel with basic options', () => {
  it('should render', () => {
    const {wrapper} = setup();
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
