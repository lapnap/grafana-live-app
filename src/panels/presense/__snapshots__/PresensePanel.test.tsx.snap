// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Presense Panel with basic options should render 1`] = `
<div>
  TODO! presense panel!!!
</div>
`;
